ExampleMod/Content/Projectiles/ExamplePaperAirplaneProjectile.cs

```diff
@@ -0,0 +1,166 @@
+			}
+
+			Projectile.velocity = newVelocity.SafeNormalize(Vector2.UnitY) * Projectile.velocity.Length(); // Set the velocity to the value we calculated above.
+
+			// If it is flying normally. i.e. not flying a loop.
+			if (!(readyForFlip && directionSameAsWind)) {
+				float yModifier = MathHelper.Lerp(0.15f, 0.05f, Math.Abs(Main.WindForVisuals));
+
+				// Half of time, decrease the y velocity a little.
+				if (Projectile.timeLeft % 40 < 20) {
+					Projectile.velocity.Y -= yModifier;
+				}
+				// The other half of time, increase the y velocity a little.
+				else {
+					Projectile.velocity.Y += yModifier;
+				}
+
+				// Cap the y velocity so the projectile falls slowly and doesn't rise too quickly
+				if (Projectile.velocity.Y < -2f) {
+					Projectile.velocity.Y = -2f;
+				}
+				if (Projectile.velocity.Y > 2f) {
+					Projectile.velocity.Y = 2f;
+				}
+
+				// Set the x velocity. MathHelper.Clamp() allows you to set a minimum and maximum value. In this case, it will always be between -6f and 6f.
+				Projectile.velocity.X = MathHelper.Clamp(Projectile.velocity.X + Main.WindForVisuals * 0.006f, -6f, 6f);
+
+				// Switch direction when the current velocity and the oldVelocity have different signs.
+				if (Projectile.velocity.X * Projectile.oldVelocity.X < 0f) {
+					Projectile.direction *= -1; // Reverse the direction
+					Projectile.ai[0] = -Main.rand.Next(120, 300); // Set Projectile.ai[0] to a random number from -120 to -599.
+					Projectile.netUpdate = true; // Sync the projectile in a multiplayer game.
+				}
+			}
+
+			// Set the rotation and spriteDirection
+			Projectile.rotation = Projectile.velocity.ToRotation();
+			Projectile.spriteDirection = Projectile.direction;
+
+			// Let's add some dust for special effect. In this case, it runs every other tick (30 ticks per second).
+			if (Projectile.timeLeft % 2 == 0) {
+				Dust.NewDustPerfect(new Vector2(Projectile.Center.X - (Projectile.width * Projectile.direction), Projectile.Center.Y), ModContent.DustType<Dusts.Sparkle>(), null, 0, default, 0.5f); //Here we spawn the dust at the back of the projectile with half scale.
+			}
+		}
+
+		// We need to do draw the projectile manually. If you don't include this, the projectile will be facing the wrong direction when flying left.
+		public override bool PreDraw(ref Color lightColor) {
+			// This is where we specify which way to flip the sprite. If the projectile is moving to the left, then flip it vertically.
+			SpriteEffects spriteEffects = ((Projectile.spriteDirection <= 0) ? SpriteEffects.FlipVertically : SpriteEffects.None);
+
+			// Getting texture of projectile
+			Texture2D texture = (Texture2D)ModContent.Request<Texture2D>(Texture);
+
+			// Get this frame on texture. If you want your projectile to be animated, see ExampleAdvancedAnimatedProjectile
+			Rectangle sourceRectangle = new Rectangle(0, 0, texture.Width, texture.Height);
+
```

Comment:
```suggestion
			// Getting texture of projectile
			Texture2D texture = Terraria.GameContent.TextureAssets.Projectile[Type].Value;

			// Get this frame on texture
			Rectangle sourceRectangle = texture.Frame(1, Main.projFrames[Type], frameY: Projectile.frame);
```

Though I would instead change `Terraria.GameContent.TextureAssets` to `TextureAssets` and add the using manually afterwards.